types/src/v0/impls/stake_table.rs

```diff
@@ -154,17 +152,19 @@ impl StaticCommittee {
     /// to be called before calling `self.stake()` so that
     /// `Self.stake_table` only needs to be updated once in a given
     /// life-cycle but may be read from many times.
-    async fn _update_stake_table(&mut self, l1_block_height: u64) {
+    async fn update_stake_table(&mut self, l1_block_height: u64) -> anyhow::Result<()> {
         let updates: StakeTables = self
-            ._provider
+            .l1_client
             .get_stake_table(l1_block_height, self._contract_address.unwrap())
```

Comment:
In order to not require a lock during network requests we should probably not combine the network request and the pure function to compute the stake table from the events here into a single function

https://github.com/EspressoSystems/espresso-sequencer/blob/816f8194ff3ca745e8bdd8cf8a0ffcfd3cc42dd3/types/src/v0/impls/l1.rs#L796-L803